### PR TITLE
Feat: Adds new tokens to `Breadcrumb`

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

This PR applies the new [Theming structure](https://github.com/vtex-sites/base.store/pull/407) to `Breadcrumb`: new scoped tokens linked to global ones.

## How does it work?

This PR uses local variables to parameterize the `Breadcrumb` and then connects these scoped tokens to the global ones.

Tokens can be [visualized in the storybook](https://nextjs-store-storybook-git-feat-theming-breadc-74b323-faststore.vercel.app/?path=/docs/molecules-breadcrumb--overview).

image

## How to test it?

- The `Breadcrumb` component and its related components should look just as it was before these changes.
- You can check the [story on Storybook](https://nextjs-store-storybook-git-feat-theming-breadc-74b323-faststore.vercel.app/?path=/docs/molecules-breadcrumb--overview).

## References

https://www.figma.com/file/zHMOtOPduiM0hNNcg21iR1/Handoff-v1(PDP)?node-id=18%3A33143

## Checklist

**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*.
- [ ] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update.

### CODE REVIEW

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  --fs-breadcrumb-dropdown-button-color               : var(--fs-color-link);
+  --fs-breadcrumb-dropdown-button-border-radius       : var(--fs-spacing-0);
+  --fs-breadcrumb-dropdown-button-transition-property : var(--fs-transition-property);
+  --fs-breadcrumb-dropdown-button-transition-timing   : var(--fs-transition-timing);
+  --fs-breadcrumb-dropdown-button-transition-function : var(--fs-transition-function);
+
+  // Dropdown Button Text
+  --fs-breadcrumb-dropdown-button-text-size           : var(--fs-text-size-3);
+  --fs-breadcrumb-dropdown-button-text-weight         : var(--fs-text-weight-bold);
+  --fs-breadcrumb-dropdown-button-text-transform      : translate(rem(1px), rem(-6px));
```
Comment (ArthurTriis1) [inline src/components/ui/Breadcrumb/breadcrumb.module.scss:45]: I think we can't use the rem() method in CSS variables definition. We are converting manually in other locals:
```suggestion
  --fs-breadcrumb-dropdown-button-text-transform      : translate(0.063rem, -0.375rem)
```

Comment (renatamottam) [reply src/components/ui/Breadcrumb/breadcrumb.module.scss:45]: This rule seems to be irrelevant to the component, could you check that?

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  [data-fs-breadcrumb-dropdown-button] {
+    display: inline;
+    padding: var(--fs-breadcrumb-dropdown-button-padding);
+    margin: var(--fs-breadcrumb-dropdown-button-margin);
+    color: var(--fs-breadcrumb-dropdown-button-color);
+    letter-spacing: var(--fs-spacing-0);
+    cursor: pointer;
+    background-color: transparent;
+    border-width: 0;
+    border-radius: var(--fs-breadcrumb-dropdown-button-border-radius);
```
Comment (renatamottam) [inline src/components/ui/Breadcrumb/breadcrumb.module.scss:143]: This line doesn't seems to be useful

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    &:visited {
+      color: var(--fs-breadcrumb-link-color-visited);
+    }
+  }
+
+  [data-fs-breadcrumb-dropdown-button] {
+    display: inline;
+    padding: var(--fs-breadcrumb-dropdown-button-padding);
+    margin: var(--fs-breadcrumb-dropdown-button-margin);
```
Comment (renatamottam) [inline src/components/ui/Breadcrumb/breadcrumb.module.scss:137]: This line doesn't seems to be useful

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    color: var(--fs-breadcrumb-dropdown-button-color);
+    letter-spacing: var(--fs-spacing-0);
```
Comment (renatamottam) [inline src/components/ui/Breadcrumb/breadcrumb.module.scss:139]: Review that usage if you change it for an `icon`

File: src/components/ui/Dropdown/dropdown.module.scss [hunk 5]
```diff
@@ -93,4 +93,8 @@
       font-size: var(--fs-dropdown-item-small-text-size);
     }
   }
+
+  [data-fs-breadcrumb-dropdown-item] [data-fs-dropdown-item-icon] {
+    margin-top: calc(-1 * var(--fs-spacing-1));
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/ui/Dropdown/dropdown.module.scss]: Agree! done [here](https://github.com/vtex-sites/nextjs.store/pull/162/commits/5a8d51d62b437a21d4652d1bd5193bd14869f06b).

File: src/components/ui/Breadcrumb/breadcrumb.module.scss (hunk 1 above)
Comment (eduardoformiga) [reply src/components/ui/Breadcrumb/breadcrumb.module.scss]: That's true! It seems irrelevant. I removed [here](https://github.com/vtex-sites/nextjs.store/pull/162/commits/e5d65ed373c5ae357861c07b2ec146dd0948b968).

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  --fs-breadcrumb-link-first-hover-background-color   : var(--fs-color-primary-bkg-light);
+
+  // Icon
+  --fs-breadcrumb-item-icon-color                     : var(--fs-color-text);
+
+  // Divider
+  --fs-breadcrumb-divider-height                      : var(--fs-spacing-3);
+  --fs-breadcrumb-divider-margin                      : var(--fs-spacing-1);
+  --fs-breadcrumb-divider-border-left-width           : var(--fs-border-width);
+  --fs-breadcrumb-divider-border-left-color           : var(--fs-border-color-light);
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/ui/Breadcrumb/breadcrumb.module.scss:31]: I don't think so. 
We are using divider on `faststore/ui` and here on the starter as well.

File: src/components/ui/Breadcrumb/breadcrumb.module.scss (hunk 2 above)
Comment (eduardoformiga) [reply src/components/ui/Breadcrumb/breadcrumb.module.scss:142]: We have this behaviour:
Screen Shot 2022-07-07 at 18 42 56

What do you think?

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+  // Dropdown Button
+  --fs-breadcrumb-dropdown-button-padding             : 0 calc(var(--fs-spacing-0) / 2);
+  --fs-breadcrumb-dropdown-button-margin              : calc(var(--fs-spacing-0) / 4);
+  --fs-breadcrumb-dropdown-button-color               : var(--fs-color-link);
+  --fs-breadcrumb-dropdown-button-border-radius       : var(--fs-spacing-0);
+  --fs-breadcrumb-dropdown-button-transition-property : var(--fs-transition-property);
+  --fs-breadcrumb-dropdown-button-transition-timing   : var(--fs-transition-timing);
+  --fs-breadcrumb-dropdown-button-transition-function : var(--fs-transition-function);
+
+  // Dropdown Button Text
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/ui/Breadcrumb/breadcrumb.module.scss]: Nice! I [changed the code](https://github.com/vtex-sites/nextjs.store/pull/162/commits/7cddf833c2992cd7e328a90939124e7ee3607a33) to use the DotsThree icon now. Could you please take a look? 👀

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+  // Link
+  --fs-breadcrumb-link-color-visited                  : var(--fs-color-link);
+
+  // Link First
+  --fs-breadcrumb-link-first-padding                  : var(--fs-spacing-1);
+  --fs-breadcrumb-link-first-border-radius            : var(--fs-border-radius-circle);
+  --fs-breadcrumb-link-first-hover-bkg-color          : var(--fs-color-primary-bkg-light);
+
+  // Icon
+  --fs-breadcrumb-item-icon-color                     : var(--fs-color-text);
```
Comment (renatamottam) [inline src/components/ui/Breadcrumb/breadcrumb.module.scss:25]: What do you think of merging those? And also, we could change
`--fs-breadcrumb-link-first`
to
`--fs-breadcrumb-link-home`

**Review by victorhmp — APPROVED** [review]

File: src/components/ui/Breadcrumb/Breadcrumb.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    token="--fs-breadcrumb-dropdown-button-transition-function"
+    value="var(--fs-transition-function)"
+  />
+</TokenTable>
+
+---
+
+## Variants
+
+### Current
```
Comment (renatamottam) [inline src/components/ui/Breadcrumb/Breadcrumb.stories.mdx:202]: I don't think this is a variant of the Breadcrumb

File: src/components/ui/Breadcrumb/breadcrumb.module.scss [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+  [data-fs-breadcrumb-link] {
+    padding: 0;
+
+    &:visited {
+      color: var(--fs-breadcrumb-link-color-visited);
+    }
+  }
+
+  [data-fs-breadcrumb-dropdown-button] {
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/ui/Breadcrumb/breadcrumb.module.scss:128]: Eagle eyes 🦅 👀 ! hahah 
Thanks, done [here](https://github.com/vtex-sites/nextjs.store/pull/162/commits/e978c500df188973b92933073485488009fde9ee).

File: src/components/ui/Breadcrumb/breadcrumb.module.scss (hunk 8 above)
Comment (eduardoformiga) [reply src/components/ui/Breadcrumb/breadcrumb.module.scss]: Nice! I rearrange the icon, merged these tokens, and renames it home. Looks better!
done [here](https://github.com/vtex-sites/nextjs.store/pull/162/commits/5784b1f240344e1217cfe4ce82728954393ae3d9).

File: src/components/ui/Breadcrumb/Breadcrumb.stories.mdx (hunk 9 above)
Comment (eduardoformiga) [reply src/components/ui/Breadcrumb/Breadcrumb.stories.mdx]: Agree with you.
I changed the implementation to deal with the current item using CSS `last-child`.
Done [here](https://github.com/vtex-sites/nextjs.store/pull/162/commits/ba67d2cf470a30a69d7a399ea6630ca51130493e).

**Pull request merged**